lib/chai/core/assertions.js

```diff
@@ -968,19 +1492,32 @@ module.exports = function (chai, _) {
   });
 
   /**
-   * ### .instanceof(constructor)
+   * ### .instanceof(constructor[, msg])
```

Comment:
During my update, I didn't really see the need to get into the details of this failing due to an invalid constructor instance. I think it's sufficiently understood by the assertion itself that it only passes when the target is an instance of the constructor, without drawing any distinction between a valid-but-different-constructor and an invalid constructor in the case of failure.